MoviesExample/app/src/main/res/layout/movies_list_item.xml

```diff
@@ -1,14 +1,29 @@
 <?xml version="1.0" encoding="utf-8"?>
-<android.support.constraint.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+
+<layout xmlns:android="http://schemas.android.com/apk/res/android">
+<data>
+  <variable
+      name="movie"
+      type="com.novoda.demo.movies.model.Movie" />
+
+  <variable
+      name="listener"
+      type="com.novoda.demo.movies.MoviesAdapter.Listener" />
+
+</data>
+
+<android.support.constraint.ConstraintLayout
   xmlns:app="http://schemas.android.com/apk/res-auto"
   xmlns:tools="http://schemas.android.com/tools"
   android:layout_width="match_parent"
-  android:layout_height="match_parent">
+  android:layout_height="wrap_content"
+  android:onClick='@{() -> listener.onMovieSelected(movie)}'>
```

Comment:
This click listener is explained here https://github.com/novoda/android-demos/pull/42/files#r191481914

Reply:
![](https://www.awesomegifs.com/wp-content/uploads/mind-blown-baby.gif)